stacks-signer/src/v0/signer_state.rs

```diff
@@ -366,10 +369,22 @@ impl LocalStateMachine {
             }
         };
 
-        // No matter what, if we're in tx replay mode, remove the tx replay set
-        // TODO: in later versions, we will only clear the tx replay
-        // set when replay is completed.
-        prior_state_machine.tx_replay_set = None;
+        if let Some(tx_replay_set) = prior_state_machine.tx_replay_set.clone() {
+            // let replay_set_hash = hash_tx_replay_set(tx_replay_set);
+            let mut hasher = DefaultHasher::new();
+            tx_replay_set.hash(&mut hasher);
+            let replay_set_hash = hasher.finish();
+            if let Ok(true) =
+                db.get_was_block_validated_by_replay_tx(signer_signature_hash, replay_set_hash)
```

Comment:
What happens to signers who don't validate block A (maybe they were offline?), where block A is the block that exhausts the replay set. If they never _process_ A, then the question doesn't really matter, because they're going to be behind chain tip anyways. But if they catch up to chain tip, they'll have processed A, but still think the replay set is active, right?

Reply:
Just to follow up with a written response - the updated logic in this PR is to have three separate branches:

- block validated with replay set, not exhausted: do nothing
- block validated with replay set, exhausted: clear replay set
- block not validated with our local replay set: clear replay set

So, to answer the question:

> What happens to signers who don't validate block A (maybe they were offline?), where block A is the block that exhausts the replay set.

When the signer processes block A, it'll hit that third branch, and their local state machine's replay set will be cleared.